### DISCUSSION TITLE
Frameworks para Screenshots

### DISCUSSION THREAD

**Discussion by inNetto** [frontendbr/forum#313] [Perguntas e Respostas]
Fala galera, uso o framework nightwatch pra fazer meus testes e2e e estou com uma grande dificuldade em conseguir configurá-lo para realizar screenshots do brownser quando meu teste falhar. Alguém aí usa nightwatch ou então algum outro framework e configura pra conseguir ver em imagem quando seu teste falha no navegador? Obrigado desde já

**Comment by felipefialho** [comment]
@IvanNetto Já tentou usar o [PhantonJS](http://phantomjs.org/) pra isso?

**Comment by inNetto** [comment]
Mas @LFeh ... esse phantomjs não é headless brownser? como eu conseguiria um screenshot do navegador já que ele não exibe interface gráfica? 

obs: te respondi agora sem muito tempo pra pesquisar legal sobre ele nesse momento ...

**Comment by fdaciuk** [comment]
Dá pra tirar screenshot com headless browser @IvanNetto =)
Mas já que está usando o nightwatch, ele tem essa opção: http://nightwatchjs.org/guide#test-settings

Não sei como está o avanço dos seus testes com o nightwatch, mas poderia ter recomendar o [Dalek](http://dalekjs.com/).

A configuração dele é bem simples, e ele tem umas features bem legais, além de não usar o selenium xD

Dá pra usar ele pra testar no browser mesmo, ou só no terminal, com phantom. E sim, dá pra tirar screenshot só com testes no terminal :stuck_out_tongue_winking_eye:

**Comment by suissa** [comment]
Nightmare eh a coisa mais simples de se usar até a minha mina já usou eheheheh.

http://www.nightmarejs.org/

**Comment by bernardodiasc** [comment]
Meu favorito https://github.com/mucsi96/nightwatch-cucumber/ :)
